### PR TITLE
SUS-762: improvements to the Image Review queue clean script.

### PR DESCRIPTION
**Pull request by harnash**
Moved image validation to a separate function and it is now used
in the Special Pages and the clean up script.

ping: @mixth-sense

### CODE REVIEW

File: extensions/wikia/ImageReview/ImageReviewHelper.class.php [hunk 1]
```diff
@@ -194,6 +194,72 @@ public function refetchImageListByTimestamp( $timestamp ) {
...
+			$sReason = 'Page is a redirect';
+		} else {
+			$oImageGlobalFile = new GlobalFile( $oImagePage );
+			if ( $oImageGlobalFile->exists() === false ) {
+				$bValidImage = false;
+				$sReason = 'File does not exist';
+			}
+		}
+
+		if ( $bValidImage === true && $oImageGlobalFile instanceof GlobalFile ) {
```
Comment (lukaszkonieczny) [inline extensions/wikia/ImageReview/ImageReviewHelper.class.php]: What do you think about moving `$oImageGlobalFile instanceof GlobalFile` check to if statement with `$oImageGlobalFile->exists() === false`?

File: extensions/wikia/ImageReview/ImageReviewHelper.class.php [hunk 2]
```diff
@@ -194,6 +194,72 @@ public function refetchImageListByTimestamp( $timestamp ) {
...
+			$sThumbUrl = $oImageGlobalFile->getUrlGenerator()
+				->width( self::IMAGE_REVIEW_THUMBNAIL_SIZE )
+				->height( self::IMAGE_REVIEW_THUMBNAIL_SIZE )
+				->thumbnailDown()
+				->url();
+
+			$aImageInfo = array(
```
Comment (lukaszkonieczny) [inline extensions/wikia/ImageReview/ImageReviewHelper.class.php]: Please change `array()` into `[ ]`

File: extensions/wikia/ImageReview/ImageReviewHelper.class.php [hunk 3]
```diff
@@ -249,90 +314,62 @@ public function getImageList( $timestamp, $state = ImageReviewStatuses::STATE_UN
...
-					];
-					continue;
-				}
-			} else {
+			if ( count( $verifiedImages ) >= self::LIMIT_IMAGES ) {
 				$unusedImages[] = $record;
+				continue;
+			}
+
+			$imageInfo = $this->checkImageValidity( $image );
```
Comment (lukaszkonieczny) [inline extensions/wikia/ImageReview/ImageReviewHelper.class.php]: `checkImageValidity` is a static method

File: extensions/wikia/ImageReview/ImageReviewTask.class.php [hunk 4]
```diff
@@ -82,6 +82,27 @@ public function delete( $pageList, $suppress = false ) {
...
+			$oDB->update(
+				'image_review',
+				[ 'state' => $aRow['state'], ],
+				[ 'wiki_id' => $aRow['wiki_id'], 'page_id' => $aRow['page_id'], ],
+				__METHOD__
+			);
+
+			WikiaLogger::instance()->info( 'ImageReviewLog', [
+				'method' => __METHOD__,
+				'message' => 'Images changed state',
```
Comment (lukaszkonieczny) [inline extensions/wikia/ImageReview/ImageReviewTask.class.php:100]: Maybe we could add field with new state value?

Comment (lukaszkonieczny) [inline extensions/wikia/ImageReview/ImageReviewTask.class.php:100]: Nevermind, I see it is in params ;)

**Comment by harnash** [comment]
Seems like those failing tests are just unstable :(

**Comment by lukaszkonieczny** [comment]
LGTM

File: extensions/wikia/ImageReview/maintenance/ImageReviewDeleteInvalidEntries.php [hunk 5]
```diff
@@ -1,19 +1,32 @@
 <?php
 
 require_once __DIR__ . '/../../../../maintenance/Maintenance.php';
+require_once __DIR__ . '/../../../../includes/wikia/nirvana/WikiaObject.class.php';
+require_once __DIR__ . '/../../../../includes/wikia/nirvana/WikiaModel.class.php';
+require_once __DIR__ . '/../../../../lib/Wikia/src/Logger/Loggable.php';
+require_once __DIR__ . '/../../../../includes/wikia/tasks/Tasks/BaseTask.class.php';
+require_once __DIR__ . '/../ImageReviewHelperBase.class.php';
+require_once __DIR__ . '/../ImageReviewHelper.class.php';
+require_once __DIR__ . '/../ImageReviewTask.class.php';
```
Comment (Grunny) [inline extensions/wikia/ImageReview/maintenance/ImageReviewDeleteInvalidEntries.php]: None of these added requires should be needed, only the Maintenance.php import. The rest of the classes will be autoloaded when used.

**Comment by ghost** [comment]
This is waiting for SUS to take another round with ImageReview.

**Pull request closed without merging**